setup.py

```diff
@@ -143,10 +149,143 @@ def _header_bad(self, path):
                 bad = target not in line
         return bad
 
-here = os.path.dirname(__file__)
-with open(os.path.join(here, 'README.rst'), 'r') as fh:
+
+# Dependency checks
+# =================
+
+# GEOS
+try:
+    geos_version = subprocess.check_output(['geos-config', '--version'])
+    geos_version = tuple(int(v) for v in geos_version.split('.'))
+    geos_includes = subprocess.check_output(['geos-config', '--includes'])
+    geos_clibs = subprocess.check_output(['geos-config', '--clibs'])
+except (OSError, ValueError, subprocess.CalledProcessError):
+    warnings.warn(
+        'Unable to determine GEOS version. Ensure you have %s or later '
+        'installed, or installation may fail.' % (
+            '.'.join(str(v) for v in GEOS_MIN_VERSION), ))
+
+    geos_includes = []
+    geos_library_dirs = []
+    if sys.platform.startswith('win'):
+        def get_config_var(name):
+            return '.'
+        geos_libraries = ['geos']
+        geos_extra_extension_args = {}
+    else:
+        geos_libaries = ['geos_c']
+        geos_extra_extension_args = dict(
```

Comment:
I wasn't sure whether `geos-config` would be available on OS X or Windows, so I left the original setup in case we'd need it there.